### ISSUE TITLE
Paypal redirect loop in widget

### ISSUE THREAD

**Issue by bobapple** [pretix/pretix#945]
I'm currently having trouble with PayPal when using the Pretix widget. I'm not sure if it is a Pretix error or a PayPal error, though. 

When using the widget to buy tickets, I select PayPal as payment method. The PayPal page opens correctly within the Lightbox/Iframe. After entering the credential, there's an infinite redirect loop (still on PayPal).

When doing the same directly on the shop page, everything works as expected. Maybe someone else has experienced the same?

**Comment by bobapple** [comment]
I just found out that when using directly the shop page, it opens a popup when using PayPal. Is this how it's intended to work?

**Comment by raphaelm** [comment]
Hmm, it *should* try to open the popup when the widget is used (or the widget has been in use before in the same browser session, we can't really tell the difference) and otherwise proceed in the same tab.

Can you check if the redirect look happens between paypal and pretix URLs or only paypal URLs?

**Comment by bobapple** [comment]
The redirect loop was with paypal links, that's why I wasn't sure if it was an error of PayPal or something with Pretix.

I worked this around by changing the `widget.js`:
```
    useIframe: function () {
        // return Math.min(screen.width, window.innerWidth) >= 800 && (this.skip_ssl || site_is_secure());
        return false;
    },
```

Today I updated to `v1.17` (from `v1.16`) and I removed the workaround to test the behaviour. Unfortunately now it's behaving different. (Event though there haven't been changes to the PayPal stuff?)

Now the lightbox is opened as expected on the same page where the widget is embedded. When selecting Paypal as payment method it tries to open a popup, which is blocked by the browser by default. So the user has to click a link to get to paypal.

Long talk short, I'm not able to reproduce the redirect loop. I will leave the `widget.js` in it's original state and see if anyone complains.

Thank you for your effort
